tests/src/whisk/core/apigw/actions/test/ApiGwRoutemgmtActionTests.scala

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.apigw.actions.test
+
+import org.junit.runner.RunWith
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.junit.JUnitRunner
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+import common.TestHelpers
+import common.TestUtils.ANY_ERROR_EXIT
+import common.Wsk
+import common.WskActorSystem
+import common.WskProps
+import common.WskTestHelpers
+import whisk.common.TransactionId
+import whisk.core.WhiskConfig
+import whisk.core.entity.Subject
+import whisk.core.entity.WhiskEntityStore
+import whisk.core.entity.WhiskAuth
+import whisk.core.entity.WhiskAuthStore
+
+/**
+ * Tests for basic CLI usage. Some of these tests require a deployed backend.
+ */
+@RunWith(classOf[JUnitRunner])
+class ApiGwRoutemgmtActionTests
+    extends TestHelpers
+    with BeforeAndAfterAll
+    with WskActorSystem
+    with WskTestHelpers {
+
+    implicit val wskprops = WskProps()
+    val wsk = new Wsk
+
+    // Need the whisk.system authentication id and pwd for invoking whisk.system private actions
+    val config = new WhiskConfig(WhiskEntityStore.requiredProperties)
+    val authstore = WhiskAuthStore.datastore(config)
+    val whiskkey = Await.result(WhiskAuth.get(authstore, Subject("whisk.system"),false)(TransactionId.unknown), 5.seconds)
+    val whisksystemAuthId = whiskkey.authkey.uuid.toString
+    val whisksystemAuthPwd = whiskkey.authkey.key.toString
+
+    override def afterAll() {
+        println("Shutting down authstore connection")
+        authstore.shutdown()
+        super.afterAll()
+    }
+
+    behavior of "API Gateway routemgmt action parameter validation"
+
+    it should "reject routemgmt actions that are invoked with not enough parameters" in {
+        val invalidArgs = Seq(
+            //getApi
+            ("/whisk.system/routemgmt/getApi", ANY_ERROR_EXIT, "namespace is required", Seq()),
```

Comment:
I included the exit code just in case we need more granular exit code validation.